packages/react-i18n/src/hooks.tsx

```diff
@@ -20,59 +17,94 @@ export function useI18n({
     );
   }
 
-  const parentI18n = React.useContext(I18nParentsContext);
-  const parentIds = parentI18n ? parentI18n.ids || [] : [];
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  const ids = React.useMemo(() => (id ? [id, ...parentIds] : parentIds), [
```

Comment:
I feel like this would have been ok as just `useMemo(() => /*stuff*/, [])`

Reply:
It could have, but for a value we know for sure that never changes it's safe to use a ref; `useMemo` is not a guarantee (I will do that for the component, too)